pkg/util/http/client.go

```diff
@@ -16,12 +18,15 @@ func (f ClientFunc) Do(req *nethttp.Request) (*nethttp.Response, error) {
 	return f(req)
 }
 
-func ClientWithBaseURL(delegate Client, baseURL *url.URL) Client {
+func ClientWithBaseURL(delegate Client, baseURL *url.URL, apiServer *config_proto.ControlPlaneCoordinates_ApiServer) Client {
```

Comment:
how about a `headers map[string]string` this way we can avoid `pkg/util` to depend on kumactl config